lib/public/Security/ISecureRandom.php

```diff
@@ -46,22 +51,39 @@ interface ISecureRandom {
 	public const CHAR_ALPHANUMERIC = self::CHAR_UPPER . self::CHAR_LOWER . self::CHAR_DIGITS;
 
 	/**
-	 * Characters that can be used for <code>generate($length, $characters)</code>, to
-	 * generate human-readable random strings. Lower- and upper-case characters and digits
-	 * are included. Characters which are ambiguous are excluded, such as I, l, and 1 and so on.
-	 *
+	 * Lowercase, uppercase characters, and digits. Ambiguous characters are excluded (e.g., I, l, and 1).
 	 * @since 23.0.0
 	 */
 	public const CHAR_HUMAN_READABLE = 'abcdefgijkmnopqrstwxyzABCDEFGHJKLMNPQRSTWXYZ23456789';
 
 	/**
-	 * Generate a random string of specified length.
-	 * @param int $length The length of the generated string
-	 * @param string $characters An optional list of characters to use if no character list is
-	 *                           specified all valid base64 characters are used.
-	 * @return string
+	 * Standard Base64 alphabet per RFC4648.
+	 * @link https://datatracker.ietf.org/doc/html/rfc4648#section-4
+	 * @since 33.0.0
+	 */
+	public const CHAR_BASE64_RFC4648 = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/';
+
+	/**
+	 * Generate a secure random string of the specified length.
+	 *
+	 * Security notes:
+	 * - For most secure applications (tokens, passwords, CSRF values), an ample and diverse
+	 *   character set, such as the default CHAR_BASE64_RFC4648, is typically a good choice.
+	 * - Overly small (<4), non-unique, or multibyte character sets weaken security and are not permitted.
+	 *
+	 * @param int $length Number of characters (must be > 0).
+	 * @param string $characters Optional list of unique, single-byte (ASCII) characters
+	 *        to use. Defaults to the CHAR_BASE64_RFC4648 alphabet. A custom set should contain at least 4
+	 *        characters, and must not contain duplicates or multibyte (non-ASCII) characters. It is strongly
+	 *        recommended to use predefined constants from ISecureRandom, which all meet the requirements.
+	 * @return string The randomly generated string.
+	 * @throws \LengthException If $length <= 0.
+	 * @throws \InvalidArgumentException if $characters contains non-ASCII characters, duplicates,
+	 *         or fewer than 4 unique characters.
 	 * @since 8.0.0
```

Comment:
```suggestion
	 * @since 35.0.0 $characters has to be at least 4 chars long, non-AS… etc
	 * @since 8.0.0
```

Add a `@since` for your change to the rules of the method